src/layout.js

```diff
@@ -434,6 +439,11 @@ export function applyStaticLayout(element) {
     if (height) {
       setStyle(element, 'height', height);
     }
+  } else if (layout == Layout.FLUID) {
+    element.classList.add('i-amphtml-layout-awaiting-size');
+    setStyle(element, 'width', width ?
```

Comment:
We can't really rely on user input, since the whole point is to alleviate the user of having to specify any sizes. Are there no other ways of accomplishing what I'm trying to do?

Reply:
I think I have an alternate approach. Instead of getting the computed style of the parent element inside layout.js, I get it inside the doubleclick impl instead. By all accounts, this appears to work. What do you think?

Reply:
Timing is important here. The earliest you can do any measurement on _the_ element is when `onLayoutMeasure` is called. If you can guarantee it - great. You definitely can't do it in the static layout, however, because an element may not be fully parsed, and the same is true about its parent.

Reply:
Btw, it's totally fine to assume input `width` to equal `auto` by default - to take the whole available space. It's implementation that we struggle with.